tasks/mds_journal_repair.py

```diff
@@ -213,16 +216,29 @@ def subtrees_assigned():
 
         # See that the second MDS will crash when it starts and tries to
         # acquire rank 1
-        self.fs.mds_restart(active_mds_names[1])
-        crasher = self.fs.mds_daemons[active_mds_names[1]].proc
-
+        crasher_id = active_mds_names[1]
+        self.fs.mds_restart(crasher_id)
         try:
-            crasher.wait()
+            self.fs.mds_daemons[crasher_id].proc.wait()
         except CommandFailedError as e:
-            log.info("MDS '{0}' crashed with status {1} as expected".format(active_mds_names[1], e.exitstatus))
-            self.fs.mds_daemons[active_mds_names[1]].proc = None
+            log.info("MDS '{0}' crashed with status {1} as expected".format(crasher_id, e.exitstatus))
+            self.fs.mds_daemons[crasher_id].proc = None
+
+            # Go remove the coredump from the crash, otherwise teuthology.internal.coredump will
+            # catch it later and treat it as a failure.
+            p = self.fs.mds_daemons[crasher_id].remote.run(args=[
+                "sudo", "sysctl", "-n", "kernel.core_pattern"], stdout=StringIO())
+            core_dir = os.path.dirname(p.stdout.getvalue().strip())
+            if core_dir:
+                log.info("Clearing core dir: {0}".format(core_dir))
+                self.fs.mds_daemons[crasher_id].remote.run(args=[
+                    "sudo", "rm", "-f", run.Raw("{0}/*".format(core_dir))
```

Comment:
I don't think we can blindly remove cores here; it might take other daemon crash information away. If we've got the core pattern and the process ID we ought to be able to identify the correct one to remove, or we can skip the delete if there are multiple files in the directory.